src/tileset/wangset/wang_tile.rs

```diff
@@ -0,0 +1,64 @@
+use std::str::FromStr;
+
+use xml::attribute::OwnedAttribute;
+
+use crate::{
+    error::Error,
+    util::{get_attrs, XmlEventResult},
+    Result, TileId,
+};
+
+/**
+The Wang ID, stored as an array of 8 u32 values.
+*/
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct WangId(pub [u32; 8]);
+
+impl FromStr for WangId {
+    type Err = Error;
+
+    fn from_str(s: &str) -> std::result::Result<WangId, Error> {
+        let mut ret = [0u32; 8];
+        let values: Vec<&str> = s
+            .trim_start_matches('[')
+            .trim_end_matches(']')
+            .split(',')
+            .collect();
+        if values.len() != 8 {
+            return Err(Error::InvalidWangIdEncoding {
+                read_string: s.to_string(),
+            });
+        }
+        for i in 0..8 {
+            ret[i] = values[i].parse::<u32>().unwrap_or(0);
+        }
+
+        Ok(WangId(ret))
+    }
+}
+
+/// Stores a tile reference along with its associated Wang ID.
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub struct WangTile {
+    /// The Wang ID,
+    pub wang_id: WangId,
+}
+
+impl WangTile {
+    /// Reads data from XML parser to create a WangTile.
+    pub fn new(
```

Comment:
```suggestion
    pub(crate) fn new(
```
Parsing functions are set to pub(crate), or at least for now. Let's change it for consistency.

Reply:
Hi, could I just commit al the changes in a single commit here from Github?, I see I can rapidly commit every change, but one by one.

Reply:
I think there's a way to stage changes and then push them all at once but I'm not sure how to do it, sorry